packages/expo/android/src/main/java/expo/modules/ReactActivityDelegateWrapper.kt

```diff
@@ -142,6 +144,29 @@ class ReactActivityDelegateWrapper(
   }
 
   override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    /**
+     * Workaround for a problem when results from [onActivityResult] are not properly delivered to modules.
+     * It happens when Android kills the [Activity] upon low memory scenario and recreates it later on.
+     *
+     * In [com.facebook.react.ReactInstanceManager.onActivityResult] you can see that if
+     * [com.facebook.react.bridge.ReactContext] is null then React would not broadcast the result to the modules
+     * and thus [expo.modules.kotlin.AppContext] would not be triggered properly.
+     *
+     * If [com.facebook.react.bridge.ReactContext] is not available when [onActivityResult] is called then
+     * let us wait for it and invoke [onActivityResult] once it's available.
+     *
+     * TODO (@bbarthec): fix it upstream?
+     */
+    if (delegate.reactInstanceManager.currentReactContext == null) {
+      val reactContextListener = object : ReactInstanceEventListener {
+        override fun onReactContextInitialized(context: ReactContext?) {
+          delegate.reactInstanceManager.removeReactInstanceEventListener(this)
+          delegate.onActivityResult(requestCode, resultCode, data)
+        }
+      }
+      return delegate.reactInstanceManager.addReactInstanceEventListener(reactContextListener)
+    }
```

Comment:
Should I add this as a CHANGELOG entry in `packages/expo`? 🤔 I'm not sure what is our approach towards `expo` package, cc @tsapeta

Reply:
it's worth to add a CHANGELOG because it fixes a real bug IMO.

Reply:
I agree with Kudo 👍